wam2layers/preprocessing/checks.py

```diff
@@ -0,0 +1,146 @@
+import warnings
+from pathlib import Path
+
+import numpy as np
+import yaml
+
+
+def _warning_on_one_line(message, category, filename, lineno, file=None, line=None):
+    """A more compact warning format.
+
+    https://stackoverflow.com/a/26433913
+    """
+    short_path = Path(filename).relative_to(Path(filename).parents[2])
+    return f'{short_path}:{lineno}: {category.__name__}: {message}\n'
+
+
+# Override the format with which warnings are printed
+warnings.formatwarning = _warning_on_one_line
+
+
+def check_monotonic_increase(data, dim="level"):
+    """Check if data is monotonically increasing in the given dimension."""
+    if not data.diff(dim).min() >= 0:
+        warnings.warn(f"Data is not monotonically increasing in the dimension {dim}.")
+
```

Comment:
Is this to check if the latitudes are increasing or decreasing?

Reply:
Not whether it is in- or decreasing, but rather that it is only, and consistently, increasing. So

* [0, 1, 2, 4, 7] would pass
* [3, 2, 1] would fail
* [1, 2, 4, 3] would fail as well